packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -542,14 +584,21 @@ function* doStartPurchaseContentFlow({
     yield* pollForPurchaseConfirmation({ contentId, contentType })
 
     // Auto-favorite the purchased item
-    if (contentType === ContentType.TRACK) {
+    if (contentType === PurchaseableContentType.TRACK) {
       yield* put(saveTrack(contentId, FavoriteSource.IMPLICIT))
     }
+    if (contentType === PurchaseableContentType.ALBUM) {
+      yield* put(saveCollection(contentId, FavoriteSource.IMPLICIT))
+    }
 
     // Check if playing the purchased track's preview and if so, stop it
     const isPreviewing = yield* select(getPreviewing)
-    const trackId = yield* select(getTrackId)
-    if (contentId === trackId && isPreviewing) {
+    const nowPlayingTrackId = yield* select(getTrackId)
+    if (
+      contentType === PurchaseableContentType.TRACK &&
+      contentId === nowPlayingTrackId &&
+      isPreviewing
+    ) {
```

Comment:
note for later - what if user is currently listening to preview of a premium track in a premium album, then purchases the album, do we want to switch out to playing the full track?

Reply:
https://linear.app/audius/issue/PAY-2612/stop-playing-preview-when-you-buy-album-containing-the-track